import { Injectable } from '@angular/core';
import { HttpClient, HttpErrorResponse } from '@angular/common/http';
import { Observable, catchError, throwError } from 'rxjs';
import { Cve } from '../models/cve.model';

@Injectable({
  providedIn: 'root',
})
export class CveService {
  private apiUrl = 'http://localhost:8080';

  constructor(private http: HttpClient) {}

  getAllCves(): Observable<Cve[]> {
    return this.http
      .get<Cve[]>(`${this.apiUrl}` + '/api/cves/')
      .pipe(catchError(this.handleError));
  }

  getCveById(id: string): Observable<Cve> {
    return this.http
      .get<Cve>(`${this.apiUrl}/api/cves/${id}`)
      .pipe(catchError(this.handleError));
}

  handleError(error: HttpErrorResponse) {
    if (error.status === 0) {
      console.error('Ha ocurrido un error:', error.error);
    } else {
      console.error(
        `El back ha devuelto el error: ${error.status}, Body: `,
        error.error
      );
    }
    return throwError(
      () => new Error('Algo ha salido mal. Por favor, intentelo m√°s tarde.')
    );
  }
}
